src/vision_overhaul.cpp

```diff
@@ -0,0 +1,455 @@
+/* General design: All characters have a vision array that takes NONE, IS_NATURAL, IS_ARTIFICIAL.
+   Vision-using code calls functions from here to see what level of vision they have.
+   We also rip out the vision mods from the current penalty function and put them here for centralization. */
+
+#include <map>
+
+#include "structs.h"
+#include "comm.h"
+#include "constants.h"
+
+bool _vision_prereqs_are_valid(struct char_data *ch, int type, const char *function_name);
+bool _vision_bit_is_valid(int bit, const char *function_name);
+
+// Do we have a given type of vision at all?
+bool has_vision(struct char_data *ch, int type, bool staff_override) {
+  if (!_vision_prereqs_are_valid(ch, type, "has_vision"))
```

Comment:
Instead of hardcoding function names, try using `__func__` and seeing if that works.